src/consul/consul_endpoint.bal

```diff
@@ -0,0 +1,329 @@
+// Copyright (c) 2018 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/http;
+
+# Consul Client object.
+# + aclToken - The acl token of the consul agent
+# + consulClient - HTTP client endpoint
+public type Client client object {
+
+    public string aclToken;
+    public http:Client consulClient;
+
+    public function __init(ConsulConfiguration consulConfig) {
+        self.consulClient = new (consulConfig.uri, config = consulConfig.clientConfig);
+        self.aclToken = consulConfig.aclToken;
+    }
+
+    # Get the details of a particular service.
+    # + serviceName - The name of the service
+    # + return - If success, returns CatalogService object with basic details, else returns error.
+    public remote function getService(string serviceName) returns @tainted CatalogService[]|error {
+        string consulPath = SERVICE_ENDPOINT + serviceName;
+
+        http:Request request = new;
+        if (self.aclToken != "") {
+            request.setHeader(CONSUL_TOKEN_HEADER, self.aclToken);
+        }
+
+        var httpResponse = self.consulClient->get(consulPath, message = request);
+        CatalogService[] serviceResponse = [];
+
+        if (httpResponse is error) {
```

Comment:
Since we are simply returning `httpResponse` if it is an error, we could maybe use `check` at L43? Will remove the if/else checks.

Reply:
Done.